packages/zendesk-adapter/src/fix_elements/fix_ticket_forms.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+import { ChangeError, ElemID, InstanceElement, isInstanceElement, ReadOnlyElementsSource } from '@salto-io/adapter-api'
+import { logger } from '@salto-io/logging'
+import _ from 'lodash'
+import { ACCOUNT_FEATURES_TYPE_NAME, ZENDESK } from '../constants'
+import { FixElementsHandler } from './types'
+
+const log = logger(module)
+const SOME_STATUSES = 'SOME_STATUSES'
+
+type Child = {
+  required_on_statuses: {
+    type: string
+    statuses?: string[] // question mark to ba able to delete it later on
+    custom_statuses: number[]
+  }
+}
+
+type InvalidTicketForm = {
+  agent_conditions: {
+    child_fields: Child[]
+  }[]
+}
+
+const isCustomStatusesEnabled = async (elementSource?: ReadOnlyElementsSource): Promise<boolean> => {
+  if (elementSource === undefined) {
+    log.error('Returning that customStatuses is disabled since no element source was provided')
+    return false
+  }
+
+  const accountFeatures = await elementSource.get(
+    new ElemID(ZENDESK, ACCOUNT_FEATURES_TYPE_NAME, 'instance', ElemID.CONFIG_NAME),
+  )
+
+  if (!isInstanceElement(accountFeatures)) {
+    log.warn('Returning that customStatuses is disabled since accountFeatures is not an instance')
+    return false
+  }
+  const customStatusesEnabled = accountFeatures.value.custom_statuses_enabled
+  if (customStatusesEnabled === undefined) {
+    log.warn('Returning that customStatuses is disabled since custom_statuses_enabled does not exist')
+    return false
+  }
+  return customStatusesEnabled.enabled
+}
+
+const invalidChild = (child: Child): boolean =>
+  _.isObject(child.required_on_statuses) &&
+  child.required_on_statuses.type === SOME_STATUSES &&
+  child.required_on_statuses.custom_statuses !== undefined &&
+  !_.isEmpty(child.required_on_statuses.custom_statuses) &&
+  child.required_on_statuses.statuses !== undefined
+
+const isInvalidTicketForm = (instanceValue: Record<string, unknown>): instanceValue is InvalidTicketForm =>
+  _.isArray(instanceValue.agent_conditions) &&
+  instanceValue.agent_conditions.some(
+    condition => _.isArray(condition.child_fields) && condition.child_fields.some(invalidChild),
+  )
+
+const returnValidInstance = (inst: InstanceElement): InstanceElement => {
```

Comment:
Maybe change the `isInvalidTicketForm` to return an assertion on the `InstanceElement`, which could simplify this:
```suggestion
const returnValidInstance = (inst: InvalidTicketForm): InstanceElement => {
```
it should save the test and the comment, no?